packages/expect/src/__tests__/matchers.test.js

```diff
@@ -247,12 +247,15 @@ describe('.toEqual()', () => {
     [new Map(), new Set()],
     [new Set([1, 2]), new Set()],
     [new Set([1, 2]), new Set([1, 2, 3])],
+    [new Set([[1], [2]]), new Set([[1], [2], [3]])],
+    [new Set([[1], [2]]), new Set([[1], [2], [2]])],
     [Immutable.Set([1, 2]), Immutable.Set()],
     [Immutable.Set([1, 2]), Immutable.Set([1, 2, 3])],
     [Immutable.OrderedSet([1, 2]), Immutable.OrderedSet([2, 1])],
     [new Map([[1, 'one'], [2, 'two']]), new Map([[1, 'one']])],
     [new Map([['a', 0]]), new Map([['b', 0]])],
     [new Map([['v', 1]]), new Map([['v', 2]])],
+    [new Map([[['v'], 1]]), new Map([[['v'], 2]])],
```

Comment:
Same as above it might be worth testing against a Map of Maps

```js
    [
      new Map([[[1], new Map([[[1], 'one']])]]),
      new Map([[[1], new Map([[[1], 'two']])]]),
    ]
```